### PR TITLE
Initial implementation of the configuration module

### LINKED ISSUE #3: Configuration module
Architecture of Configuration module(what it should contain?), configuration via environment variables

### PR DESCRIPTION
**Pull request by dkiadrynski**
* Added ability to use configuration files in YAML format.
* Added ability to use configuration files in TOML format.
* Added ability to upload all the configuration files in a folder in different formats using `DirectoryProbing` functionality.
* Added ability to set up this configuration module using `Mafin.Configuration.json` file.
* Added ability to load environment-dependent configuration from files.
* Added the `ConfigurationSectionAttribute` attribute and the appropriate extension method to specify the root to configuration model.
* Added examples to the `README` file.

### CODE REVIEW

File: src/Mafin.Configuration/Providers/SettingsFile/SettingsFileConfigurationBuilderExtension.cs [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+namespace Mafin.Configuration.Providers.SettingsFile;
+
+/// <summary>
+/// Extension methods for <see cref="IConfigurationBuilder"/> to add <see cref="SettingsFileConfigurationSource"/>.
+/// </summary>
+public static class SettingsFileConfigurationBuilderExtension
+{
+    private static string DefaultSettingsFile =>
+        Path.Combine(Environment.CurrentDirectory, "Mafin.Configuration.json");
```
Comment (oleksandr-fomenko) [inline src/Mafin.Configuration/Providers/SettingsFile/SettingsFileConfigurationBuilderExtension.cs]: What will happen if the "Mafin.Configuration.json" file is not created? Suggesting extracting the "Mafin.Configuration.json" value into a constant.

File: src/Mafin.Configuration/Mafin.Configuration.csproj [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  <ItemGroup>
+    <InternalsVisibleTo Include="$(MSBuildProjectName).Tests.Unit" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Configuration" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Ini" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Xml" />
```
Comment (oleksandr-fomenko) [inline src/Mafin.Configuration/Mafin.Configuration.csproj:21]: please add support of the 
Microsoft.Extensions.Configuration.EnvironmentVariables
Microsoft.Extensions.Configuration.UserSecrets

File: src/Mafin.Configuration/build/Mafin.Configuration.props [hunk 3]
```diff
@@ -0,0 +1,17 @@
+<Project ToolsVersion="12.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <ItemGroup>
```
Comment (oleksandr-fomenko) [inline src/Mafin.Configuration/build/Mafin.Configuration.props:2]: do we really need to handle all of these file names? especially the fact that src/Mafin.Configuration/Providers/SettingsFile/SettingsFileConfigurationBuilderExtension.cs line 12 has a hardcoded file name.

**Comment by oleksandr-fomenko** [comment]
Do we really need TOML and YAML formats for being supported?
I might be wrong (or missed a meeting where it's been discussed), but the initial idea was to support JSON format only.

**Comment by oleksandr-fomenko** [comment]
Any plans to manage secret values? E.g provide encryption/decryption of the secret values

**Comment by dkiadrynski** [comment]
@oleksandr-fomenko

> Do we really need TOML and YAML formats for being supported? I might be wrong (or missed a meeting where it's been discussed), but the initial idea was to support JSON format only.

The initial requirements (you can find them in MAFIN ideas file) we discussed at the sync were follow:

```
# Take configuration from different sources: 
1. Files: 
1.1 JSON 
1.2 XML 
1.3 YAML 
1.4 TOML 
1.5 INI 
1.6 Etc. 
2. Environment variables 
3. User Secrets #
4. Key vaults 

# Take implementation details from Microsoft configuration

# Fluent Builder? 

# Change config on runtime? 

# Extensibility (custom sources, e.g. SQL)? 
```

> Any plans to manage secret values? E.g provide encryption/decryption of the secret values

Just thinking about KeyVaults so far

File: src/Mafin.Configuration/Providers/SettingsFile/SettingsFileConfigurationBuilderExtension.cs (hunk 1 above)
Comment (dkiadrynski) [reply src/Mafin.Configuration/Providers/SettingsFile/SettingsFileConfigurationBuilderExtension.cs]: > What will happen if the "Mafin.Configuration.json" file is not created?
```
---- System.IO.FileLoadException : Unable to load configuration from Mafin.Configuration.json
```

File: .github/dependabot.yml [hunk 4]
```diff
@@ -7,4 +7,4 @@ updates:
 - package-ecosystem: "nuget"
   directory: "/"
   schedule:
-    interval: "weekly"
+    interval: "weekly"
```
Comment (Molnix888) [inline .github/dependabot.yml]: Could you please, setup editorconfig recognition in the IDE and revert such changes?

**Review by kozlydiy — DISMISSED** [review]

**Review by Molnix888 — APPROVED** [review]

**Pull request merged**